### PR TITLE
Supplier of `StorageFactory` in `BoundedContext`

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, I've merged changes. PTAL.

File: server/src/test/java/org/spine3/util/EnvironmentShould.java [hunk 1]
```diff
@@ -94,40 +94,46 @@ public void tell_that_we_are_under_tests_if_env_var_set_to_true() throws Excepti
...
     @Test
     public void tell_that_we_are_not_under_tests_if_env_set_to_something_else() {
         System.setProperty(Environment.ENV_KEY_TESTS, "neitherTrueNor1");
 
         assertFalse(environment.isTests());
+        assertTrue(environment.isProduction());
     }
 
     @Test
     public void turn_into_tests_mode() {
```
Comment (armiol) [inline server/src/test/java/org/spine3/util/EnvironmentShould.java]: This name should be updated according to the changes made in #289.

File: server/src/test/java/org/spine3/util/EnvironmentShould.java [hunk 2]
```diff
@@ -94,40 +94,46 @@ public void tell_that_we_are_under_tests_if_env_var_set_to_true() throws Excepti
...
         environment.setToTests();
 
         assertTrue(environment.isTests());
+        assertFalse(environment.isProduction());
     }
 
     @Test
     public void turn_into_production_mode() {
```
Comment (armiol) [inline server/src/test/java/org/spine3/util/EnvironmentShould.java]: This name should be updated according to the changes made in #289.

Comment (alexander-yevsyukov) [inline server/src/test/java/org/spine3/util/EnvironmentShould.java]: @armiol, fixed. PTAL.

**Pull request merged**